xds/internal/xdsclient/xds.go

```diff
@@ -561,6 +568,40 @@ func routesProtoToSlice(routes []*v3routepb.Route, logger *grpclog.PrefixLogger,
 	return routesRet, nil
 }
 
+func hashPoliciesProtoToSlice(policies []*v3routepb.RouteAction_HashPolicy) ([]*HashPolicy, error) {
+	var hashPoliciesRet []*HashPolicy
+	for _, p := range policies {
+		// Hash Policies are only applicable for a Ring Hash LB.
+		if !env.RingHashSupport {
+			return nil, errors.New("hash policies are only applicable for ring hash lb policy")
+		}
+		var policy HashPolicy
+		policy.Terminal = p.Terminal
```

Comment:
Nit: this can be oneline
`policy := HashPolicy{Terminal: p.Terminal}`

Reply:
Done.